packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart

```diff
@@ -704,9 +707,9 @@ class SectionLabelPainter extends FlameChartPainter {
   }
 
   @override
-  bool shouldRepaint(SectionLabelPainter oldDelegate) {
-    return verticalScrollOffset != oldDelegate.verticalScrollOffset ||
-        eventGroups != oldDelegate.eventGroups ||
+  bool shouldRepaint(CustomPainter oldDelegate) {
+    return oldDelegate is SectionLabelPainter &&
+            eventGroups != oldDelegate.eventGroups ||
```

Comment:
formatting?

Reply:
switched to an if as this logic was actually wrong.